libminifi/test/unit/ZlibStreamTests.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <utility>
+#include <string>
+#include <random>
+#include <algorithm>
+#include "../TestBase.h"
+#include "io/ZlibStream.h"
+#include "utils/StringUtils.h"
+
+namespace io = org::apache::nifi::minifi::io;
+
+TEST_CASE("gzip compression and decompression", "[basic]") {
+  /* Compression*/
+  io::ZlibCompressStream compressStream;
+
+  std::string original;
+  SECTION("Empty") {
+  }
+  SECTION("Simple content in one write") {
+    REQUIRE(-1 != compressStream.write(reinterpret_cast<uint8_t*>(const_cast<char*>("foobar")), strlen("foobar")));
+    original += "foobar";
+  }
+  SECTION("Simple content in two writes") {
+    REQUIRE(-1 != compressStream.write(reinterpret_cast<uint8_t*>(const_cast<char*>("foo")), strlen("foo")));
+    original += "foo";
+    REQUIRE(-1 != compressStream.write(reinterpret_cast<uint8_t*>(const_cast<char*>("bar")), strlen("bar")));
```

Comment:
I think the return values should match the length of the string. 
Or in case they return the number of compressed bytes written, let's check for grearter than zero.

Reply:
It must return the number of bytes consumed, will change to an exact check.